documentation/processing.tex

```diff
@@ -1 +1,315 @@
+	
+	\item[\texttt{logic\_expression}] Represents a logic expression. Note, logic, and arithmetic expression have some common functions, but their behavior differs. 
+	
+	\item[\texttt{arithmetic\_expression}] Represents an arithmetic expression.
+	
+	\item[\texttt{arithmetic\_logic\_expr\_wrapper}] HLASM language supports expressions with operands of mixed types. For more straightforward and readable use of arithmetic and logical expression, this class wraps these expression types under one class.
+	
+	\item[\texttt{character\_expression}] Represents a character expression.
+	
+	\item[\texttt{ebcdic\_encoding}] This class defines a custom EBCDIC literal and provides helper functions for conversion between EBCDIC and ASCII. EBCDIC is a character encoding used in the IBM mainframe. It has a different layout than ASCII (see \cref{ca_expr:ebcdic}).
+	
+	\item[\texttt{error\_messages}] It is a static class with list of all \texttt{diagnostic\_op} generated from expressions.
+	
+	
+\end{description}
+\begin{lstlisting}[label={ca_expr:example},language=C++,backgroundcolor=\color{cyan!10}, captionpos=b, caption=Defition of \texttt{copy\_return\_on\_error} macro and an example of its usage while evaluating unary expression \texttt{B2A("123")}. \texttt{this} is an \texttt{character\_expression} with value \texttt{"123"}.]
+#define copy_return_on_error(arg, type) \
+do \
+{ \
+auto te = test_and_copy_error<type>(arg); \
+if (te != nullptr) return te; \
+} \
+while(0)
+
+//before upward propagating the operation result
+//call the macro
+copy_return_on_error(this, arithmetic_expression);
+return arithmetic_expression::from_string(value_, 2);
+
+\end{lstlisting}
+
+\begin{figure}
+	\centering
+	\includegraphics[height=13cm]{img/ebcdic}
+	\caption{EBCDIC layout. Taken from \url{https://i.stack.imgur.com/h3u5A.png}.}
+	\label{ca_expr:ebcdic}
+\end{figure}
+
+\subsubsection{CA expression evaluation}
+\label{ca_expr:eval}
+
+In the previous section, we described the representation of the CA expressions themselves. In this section, we explain the coupling of CA expressions with grammar via visitor. 
+
+The \texttt{expression\_evaluator} encapsulates the coupling logic between the grammar and the expression logic. That is, the evaluator has a notion about grammar, which translates into C++ expression logic. At this point, it is essential for the reader to be familiar with the grammar (see \cref{parser:rules}).
+
+The top-level expression first gathers a list of space-separated expressions. We gather this list because a token may be a keyword (such as \texttt{AND} operator) or variable identifier, depending on a position in an expression (using language keywords as identifiers is allowed in HLASM). \texttt{expression::evaluate} provides the disambiguation (see \cref{ca_expr:logic}). 
```

Comment:
takze ten list zbierame iba preto ze token moze byt keyword alebo variable identifier? a nie preto aby sme mohli evaluovat expressny?

v sekcii ca_expr je pisane nieco o disambiguation medzi variable identifier a keyword?

Reply:
tak disambiguation sa robi pocas evaluation, ak som si spravne vedomi, tak inak a inde sa to robit neda

Reply:
mne pride zvlastna ta veta ked hovori ze ten list sa zbiera iba kvoli tomu aby sa robil disambiguation...